src/core/utils.ts

```diff
@@ -35,13 +36,13 @@ export function trimLeft(str: string, char: string): string {
 
 /**
  * Returns whether the given file path points to a location inside the given
- * directory.
+ * directory. Also return true if the paths are the same.
  */
-export function isPathInside(directory: string, filePath: string): boolean {
-  if (process.platform === 'win32') {
-    return filePath.toLowerCase().startsWith(directory.toLowerCase());
-  }
-  return filePath.startsWith(directory);
+export function isPathInside(directory: string, file: string): boolean {
+  // If the path from the directory to the file does not require traversing up,
+  // then it must be either a descendent or the same directory. Note this is
+  // case-insensitive on Windows (which is what we want).
+  return !pathlib.relative(directory, file).startsWith('..');
```

Comment:
A small issue that probably won't come up:

I think `isPathInside('c:\\foo', 'd:\\bar')` returns true right now. Maybe we should add an extra check if we're on windows?

Reply:
Good catch. I've just switched to pull in path-is-inside, which already handles this case correctly. Note that it has the parameters in reverse order (which is the more intuitive order to me anyway). Also filed https://github.com/domenic/path-is-inside/pull/7 to add typings there, but added custom ones here for now.